lib/marginalia/comment.rb

```diff
@@ -19,12 +30,19 @@ def self.update_adapter!(adapter)
       self.marginalia_adapter = adapter
     end
 
+    # @param [String] value
+    # @return [String]
+    def self.quote_value(value)
+      quote_values ? "'#{value.gsub("'", "\\\\'")}'" : value
+    end
+
     def self.construct_comment
       ret = String.new
       self.components.each do |c|
         component_value = self.send(c)
         if component_value.present?
-          ret << "#{c}:#{component_value},"
+          ret << "#{c}#{key_value_separator}"\
+                 "#{quote_value(component_value)},"
```

Comment:
Suggest introducing a configurable `formatter` block rather than separate low-level config points, including `:default` and `:sqlcommenter` shorthands for "standard" formatters.